src/auditor/dynamic_linkage.jl

```diff
@@ -1,5 +1,30 @@
 using ObjectFile.ELF
 
+function os_from_elf_note(oh::ELFHandle)
+    for section in Sections(oh)
+        section_type(section) == ELF.SHT_NOTE || continue
+        seek(oh, section_offset(section))
+        name_length = read(oh, UInt32)
+        iszero(name_length) && continue
+        descriptor_length = read(oh, UInt32)
+        note_type = read(oh, UInt32)
+        name = String(read(oh, name_length - 1))  # skip trailing NUL
+        if note_type == 1
+            # Technically it's part of the Linux specification that any executable should
+            # have an ELF note with type 1, name GNU, and descriptor length ≥4, but in
+            # practice I haven't observed that consistently, especially on musl. So for
+            # now, only bother checking FreeBSD, which uses an ELF note rather than OS/ABI
+            # to identify itself on AArch64 and RISC-V.
+            if name == "FreeBSD" && descriptor_length == 4
+                return name
+            end
+        end
+    end
+    return nothing
+end
+
+os_from_elf_note(::ObjectHandle) = nothing
+
```

Comment:
Good to know a blank line was not covered by tests!

![image](https://github.com/user-attachments/assets/10ce54e9-39d9-453e-a5d9-7840e3930134)

Reply:
```julia
@test isempty(readlines("src/auditor/dynamic_linkage.jl")[27])
```